### PR TITLE
PSK-3: Merge PSK-3-be-development-for-psk-1-true-up-data-model-with-pegasus-core to PSP-26-true-up-data-model

### PR DESCRIPTION
**Pull request by acs-jira**

### CODE REVIEW

File: acs-core/src/main/java/com/arrow/acs/client/model/AccessKeyModel.java [hunk 1]
```diff
@@ -9,46 +9,118 @@
...
  *******************************************************************************/
 package com.arrow.acs.client.model;
+import java.util.ArrayList;
+import java.util.List;
 
-import java.io.Serializable;
-
-public class AccessKeyModel implements Serializable {
+public class AccessKeyModel extends ModelAbstract<AccessKeyModel> {
     private static final long serialVersionUID = 2009328107616383251L;
```
Comment (dantonov-arrow) [inline acs-core/src/main/java/com/arrow/acs/client/model/AccessKeyModel.java]: In general, it is desired to re-generate serialVersionUID if change class fields

File: acs-core/src/main/java/com/arrow/acs/client/model/AccessLevel.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package com.arrow.acs.client.model;
```
Comment (dantonov-arrow) [inline acs-core/src/main/java/com/arrow/acs/client/model/AccessLevel.java:11]: please add copyright

File: acs-core/src/main/java/com/arrow/acs/client/model/AccessPrivilegeModel.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package com.arrow.acs.client.model;
```
Comment (dantonov-arrow) [inline acs-core/src/main/java/com/arrow/acs/client/model/AccessPrivilegeModel.java:11]: please add copyright

File: acs-core/src/main/java/com/arrow/acs/client/model/AccessPrivilegeModel.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+
+public class AccessPrivilegeModel {
```
Comment (dantonov-arrow) [inline acs-core/src/main/java/com/arrow/acs/client/model/AccessPrivilegeModel.java]: please add implements Serializable and generate serialVersionUID

File: acs-core/src/main/java/com/arrow/acs/client/model/ApplicationModel.java [hunk 5]
```diff
@@ -10,6 +10,9 @@
  *******************************************************************************/
 package com.arrow.acs.client.model;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import com.fasterxml.jackson.annotation.JsonIgnore;
 
 public class ApplicationModel extends DefinitionModelAbstract<ApplicationModel> {
```
Comment (dantonov-arrow) [inline acs-core/src/main/java/com/arrow/acs/client/model/ApplicationModel.java:18]: In general, it is desired to re-generate serialVersionUID if change class fields

File: acs-core/src/main/java/com/arrow/acs/client/model/AuditLogModel.java [hunk 6]
```diff
@@ -16,6 +16,7 @@
 public class AuditLogModel extends TsModelAbstract<AuditLogModel> {
 	private static final long serialVersionUID = -6713413123784260777L;
```
Comment (dantonov-arrow) [inline acs-core/src/main/java/com/arrow/acs/client/model/AuditLogModel.java]: In general, it is desired to re-generate serialVersionUID if change class fields

File: acs-core/src/main/java/com/arrow/acs/client/model/CompanyModel.java [hunk 7]
```diff
@@ -10,6 +10,9 @@
  *******************************************************************************/
 package com.arrow.acs.client.model;
 
+import java.util.HashSet;
+import java.util.Set;
+
 public class CompanyModel extends AuditableDocumentModelAbstract<CompanyModel> {
 	private static final long serialVersionUID = -1173608869956123803L;
```
Comment (dantonov-arrow) [inline acs-core/src/main/java/com/arrow/acs/client/model/CompanyModel.java]: In general, it is desired to re-generate serialVersionUID if change class fields

File: acs-core/src/main/java/com/arrow/acs/client/model/CompanyModel.java [hunk 8]
```diff
@@ -23,6 +26,7 @@
 	private PasswordPolicyModel passwordPolicy;
 	private LoginPolicyModel loginPolicy;
 	private String parentCompanyHid;
+	private Set<String>   impersonateUsers = new HashSet<>();;
```
Comment (dantonov-arrow) [inline acs-core/src/main/java/com/arrow/acs/client/model/CompanyModel.java]: please format this line - a lot of spaces before variable name
please remove last ";"

File: acs-core/src/main/java/com/arrow/acs/client/model/ContactModel.java [hunk 9]
```diff
@@ -24,6 +24,7 @@
 	private String cell;
 	private String fax;
 	private String monitorExt;
+	private String  middleName;
```
Comment (dantonov-arrow) [inline acs-core/src/main/java/com/arrow/acs/client/model/ContactModel.java:27]: In general, it is desired to re-generate serialVersionUID if change class fields

File: acs-core/src/main/java/com/arrow/acs/client/model/ProductFeatureModel.java [hunk 10]
```diff
@@ -0,0 +1,48 @@
+package com.arrow.acs.client.model;
```
Comment (dantonov-arrow) [inline acs-core/src/main/java/com/arrow/acs/client/model/ProductFeatureModel.java]: Please add copyright

File: acs-core/src/main/java/com/arrow/acs/client/model/ProductFeatureModel.java [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+public class ProductFeatureModel {
```
Comment (dantonov-arrow) [inline acs-core/src/main/java/com/arrow/acs/client/model/ProductFeatureModel.java]: according to document from story, the name of this class should be" ProductFeature"

File: acs-core/src/main/java/com/arrow/acs/client/model/ProductFeatureModel.java [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+
```
Comment (dantonov-arrow) [inline acs-core/src/main/java/com/arrow/acs/client/model/ProductFeatureModel.java]: please add implements Serializable and generate serialVersionUID

File: acs-core/src/main/java/com/arrow/acs/client/model/ProductModel.java [hunk 13]
```diff
@@ -18,6 +21,7 @@
 	private String systemName;
 	private boolean apiSigningRequired;
 	private String parentProductHid;
+	private List<ProductFeatureModel> features= new ArrayList<>();
```
Comment (dantonov-arrow) [inline acs-core/src/main/java/com/arrow/acs/client/model/ProductModel.java]: please format this line

File: acs-core/src/main/java/com/arrow/acs/client/model/ProductModel.java [hunk 14]
```diff
@@ -18,6 +21,7 @@
 	private String systemName;
```
Comment (dantonov-arrow) [inline acs-core/src/main/java/com/arrow/acs/client/model/ProductModel.java:21]: In general, it is desired to re-generate serialVersionUID if change class fields

**Review by dantonov-arrow — COMMENTED** [review]
added comments

File: acs-core/src/main/java/com/arrow/acs/client/model/SocialEventModel.java [hunk 15]
```diff
@@ -1,95 +1,95 @@
-/*******************************************************************************
```
Comment (ingladk) [inline acs-core/src/main/java/com/arrow/acs/client/model/SocialEventModel.java:1]: SocialEventModel should be moved to Pegasus

File: acs-core/src/main/java/com/arrow/acs/client/model/RoleModel.java [hunk 16]
```diff
@@ -33,7 +33,8 @@
 	protected RoleModel self() {
 		return this;
 	}
-
+
```
Comment (ingladk) [inline acs-core/src/main/java/com/arrow/acs/client/model/RoleModel.java]: Please apply code formatting rules and remove empty lines

File: acs-core/src/main/java/com/arrow/acs/client/model/RoleModel.java [hunk 17]
```diff
@@ -72,6 +89,13 @@ public RoleModel withEditable(boolean editable) {
 		setEditable(editable);
 		return this;
 	}
+	
+
```
Comment (ingladk) [inline acs-core/src/main/java/com/arrow/acs/client/model/RoleModel.java]: Please apply code formatting rules and remove empty lines

File: acs-core/src/main/java/com/arrow/acs/client/model/UserAuthModel.java [hunk 18]
```diff
@@ -0,0 +1,65 @@
+package com.arrow.acs.client.model;
+
+public class UserAuthModel {
```
Comment (ingladk) [inline acs-core/src/main/java/com/arrow/acs/client/model/UserAuthModel.java]: Please implement Serializable and add generated serialVersionUID

File: acs-core/src/main/java/com/arrow/acs/client/model/UserModel.java [hunk 19]
```diff
@@ -23,6 +23,8 @@
 	private String companyHid;
 	private ContactModel contact;
 	private AddressModel address;
+	private Long accountLockTimeout;
```
Comment (ingladk) [inline acs-core/src/main/java/com/arrow/acs/client/model/UserModel.java:26]: Please re-generate serialVersionUID

File: acs-core/src/main/java/com/arrow/acs/client/model/UserAuthModel.java [hunk 20]
```diff
@@ -0,0 +1,65 @@
+package com.arrow.acs.client.model;
```
Comment (ingladk) [inline acs-core/src/main/java/com/arrow/acs/client/model/UserAuthModel.java:11]: Please add copyright

**Pull request merged**